### ISSUE TITLE
US Date Format Showing As UK Date Format

### ISSUE THREAD

**Issue by 420Banners** [revive-adserver/revive-adserver#983]
The main page advertiser-index.php date for US has the year at the beginning like this: 
2018-02-24

The stats.php page for the client the date for US is showing as UK format with day first, then month, then year like this: 
24-02-2018

The advertiser-campaigns.php page is also showing as UK format with day first, then month, then year like this: 
24-02-2018

All of these areas should have the date displayed as 02-24-2018

How do we do this, until you get this bug sorted?

Thank you.

**Comment by 420Banners** [comment]
How's this coming along guys, any chance we can get an update after almost 2 weeks? :) 
Thank you.

**Comment by erikgeurts** [comment]
US date format is one of dozens of different ways of representing dates in the various locales around the globe. While it is true that this might be confusing, it's hardly a bug that causes the application to come crashing down. As such, we prefer to give priority to other issues first.

**Comment by 420Banners** [comment]
I can see that perspective, how about this perspective.

This is an "AdServer" developed to sell advertising and keep advertisers who pay our bills happy.

This issue is causing confusion and loss of revenue, from the following reasons.

1. Advertisers are getting emails about expiring ad campaigns, but not realizing the urgency of the date, because at a quick glance it looks as if it could be expiring 6 months from now, instead of 30 days. This causes a loss in traffic for the advertisers when expired, and a loss of revenue for us, the service provider.

2. Advertisers are getting confused when trying to view stats, resulting in confusion, anger and in some cases even calling us lazy and unprofessional, because the date format is wrong and they don't understand. Also another loss of revenue, due to the client not renewing the ad campaign, due to too much unprofessional confusion.

3. Causing confusion with our webmaster who posts the ads for our advertisers, resulting in incorrect dates being entered. Again, causing more confusion and frustration from not only staff, but clients as well. Also another loss of revenue, due to the client not renewing the ad campaign, due to too much unprofessional confusion.

I realize you guys are based out of the UK and that you may see the US as your last priority, however I know we're not the only US clients who have been using this software in the US for over a decade. This issue has been going on for years, can you please either...

1. Fix it now and release and update

or 

2. Provide me with the info needed for my programmer to edit your software until then.

Thank you very much, our revenue stream is counting on you, as we have for over a decade.

**Comment by mbeccati** [comment]
Hi @420Banners, thanks for your report. I apologise for not checking thoroughly, but I was sure we had both the UK and US english variants available since a long time and I couldn't understand what the complaint was about. I have prepared pull request #988, which we'll merge as soon as we verify that our Crowdin integration isn't going to break because of it. In the meantime, it shoud be fairly easy for you to create the new `lib/max/language/en_US` direcotry and copy the two files over there. Then just switch the language in the account preferences and you're set.

**Comment by erikgeurts** [comment]
We, the Revive Adserver project team, are not based in the UK. We're a team with people in The Netherlands, Italy, and Australia. We are very much aware of the fact that there are many users of our software in the US.

In addition, you are not a 'client' of ours, you simply downloaded and installed a free, open source application and have been using it to run your business. Well done! That really is the goal we have, to make and maintain software that people and business think is valuable and useful. We work on this project as volunteers, in our own time, all of us use the income generated with other activities to enable us to work on Revive Adserver, and we've all been doing that for many years now.

Some of the users of the software feel that it is important that the tool they use to run their ad business is maintained, bugs are fixed, and development continues. For that reason, we decided to create a sponsoring option, so that those users can contribute a small portion of their revenue to the project and help us pay for some of the bills. See https://www.patreon.com/revive_adserver for more information. You can contribute as much or as little as you want, many sponsors pledged $10 a month for example.

We use the money raised in this way to pay for things like servers to host our website, forum, documentation, sync server, domain names, SSL certificates, and so on, and also to hire outside help, to pay the accountant for the yearly report, bank fees, and many other things. As a business owner, you probably know better than anyone what's involved in just the basics of an organisation, and you might be surprised about the costs for running an open source project that has many thousands of users world wide.

In summary: Revive Adserver is free, open source software, and it will remain free. If you use it to run a commercial business or activity and generate revenue from that, it's yours to keep. However, if you feel that the software is important to you now and in the future, it might be worth considering becoming a patron.

**Comment by 420Banners** [comment]
mbeccati, thank you for taking on the bug fixes, we are truly grateful.

Is this already done it appears, will it be in the next update?

erikgeurts, with OpenX, their options were FREE or $10,000 a month, there was no middle ground. 

We will gladly donate funds to your project, once we get our long term issues resolved.

Thank you very much, we are truly grateful for you.

**Comment by mbeccati** [comment]
@420Banners possibly, yes, it will be in the next release.

OpenX Inc. hasn't had anything to do with the project since almost 5 years now ;)

**Comment by 420Banners** [comment]
@mbeccati I realize this.

If I had known before, we would have been donating already. :)

**Comment by 420Banners** [comment]
@mbeccati may I ask why you closed this, without there being a new update in place with this fix?

Thank you :)

**Comment by andrewatfornax** [comment]
Hi @420Banners - the ticket is closed, because a fix has been coded, and has been merged into the main branch of the code, ready for release in Revive Adserver 4.2.0. We (like most projects) close tickets at that point, otherwise we'd have loads of open tickets that we would then have to manage when the release comes out.

4.2.0 will be out in due course, with the fix, but you are of course welcome to take the changes from the fix and apply to your current system if you need that change in place right away.

Thanks.

**Comment by 420Banners** [comment]
Thank you @andrewatfornax 

I absolutely LOVE the new version number.

How ironic after all this time, the update I've been waiting for. :)

Thank you gentlemen!

**Comment by 420Banners** [comment]
PS how do we get notified updates when they are released?

**Comment by andrewatfornax** [comment]
New releases are announced on our blog, via an email list you can subscribe to, Twitter, Facebook and Patreon.

Certain levels of Patreon supporters also get advance notice of the release date!

**Comment by erikgeurts** [comment]
Sign up for an email notification sent every time a new version is released at https://www.revive-adserver.com/mailinglist/

**Comment by erikgeurts** [comment]
Aso, to join our Patreon program and contribute a small fraction of your monthly ad revenue to the project, visit https://www.patreon.com/revive_adserver

**Comment by 420Banners** [comment]
Thank you @erikgeurts that's what I needed :)

**Comment by 420Banners** [comment]
Our programmer says the date format did not change, he says...

mbeccati's instructions are already in place as shown in the attached screenshot from the new version -
https://github.com/revive-adserver/revive-adserver/issues/983#issuecomment-400943340

account-preferences-timezone.php is set to America/Los_Angeles - https://www.xxx.com/bmg/www/admin/account-preferences-timezone.php

Unless there should be a new option to select under account-preferences-timezone.php for the change to go into effect?

**Comment by mbeccati** [comment]
Did you change the language preferences to English (Us)?

**Comment by 420Banners** [comment]
As stated above...

account-preferences-timezone.php is set to "America/Los_Angeles" 

There is no option for "English (Us)" in the drop down menu on that page.

Can you please be more specific?

Thank you.

**Comment by 420Banners** [comment]
Unfortunately, it appears to be resolved in the admin control panel, however they missed updating the email that goes out advising of expiration.

Example for an August expiration email notification

The campaign belonging to XXX shown below is due to end on 01-08-2019.

This should read 08-01-2019 for US format

Is there a template somewhere we can edit or something?

Please help, we've been trying to get this resolved for over a year.

Thank you.

**Comment by andrewatfornax** [comment]
I've just done some other coding related to formatting of dates & times, so I'll pick this one up next.

**Comment by 420Banners** [comment]
Thank you Andrew, we are truly grateful.

**Comment by andrewatfornax** [comment]
So, it's the code in lib/OA/Email.php that handles the sending of email-based alerts and reports, and it's the sendCampaignImpendingExpiryEmail() method that handles the sending of alerts that a campaign is about to expire.

This method handles two different reasons for expiration - either the campaign is about to reach its impressions target, or the end date is approaching. It's the latter of these two that we're talking about here in terms of the way the end date is formatted in the email.

The method delegates the preparation of that part of the report to another method called prepareCampaignImpendingExpiryEmail(), however, on line https://github.com/revive-adserver/revive-adserver/blob/master/lib/OA/Email.php#L608, it looks to me like the code is already attempting to format the date as it is passed into the prepareCampaignImpendingExpiryEmail() method.

So, in terms of an immediate fix, you could hack this file and just set the date format here.

In terms of a proper fix, I need to figure out why that's not already working, because there's definitely code there to format the date!

**Comment by 420Banners** [comment]
@andrewatfornax Can you let me know what the edit is to make this happen right away? Thanks!

**Comment by andrewatfornax** [comment]
@420Banners - set the string to the same as in the en_US language pack: https://github.com/revive-adserver/revive-adserver/blob/master/lib/max/language/en_US/default.lang.php#L17

**Comment by 420Banners** [comment]
Marc,

 

Why is this closed, did they fix the problem or not?

https://github.com/revive-adserver/revive-adserver/issues/983

 

Greenest Regards,

Webmaster

 

420 Magazine

PO Box 3420

Hollywood, CA 90078

http://www.420magazine.com <http://www.420magazine.com/> 

http://www.twitter.com/420magazine

http://www.youtube.com/420magazine 

http://www.facebook.com/420magazine

http://www.instagram.com/420magazine 

 

 

 

Confidentiality Notice. This e-mail is covered by the Electronic Communications Privacy Act, 18 U.S.C. Sections 2510-2521 and is legally privileged. This e-mail is the property of 420 Magazine, Inc. and contains confidential and/or privileged information. This e-mail is intended to be received and reviewed only by the individual named above. Any review, use, dissemination, disclosure or copying of this e-mail or any attachments is strictly prohibited. If you have received this e-mail in error, please immediately notify the sender at support@420magazine.com and delete all copies of the e-mail and any attachments. Thank you for your cooperation and assistance.

**Comment by 420Banners** [comment]
We upgraded the software which was said to include a fix to the issue but we're still encountering the problem. The language is set to English (US) and timezone to America/Los_Angeles, but ad expiration reminder emails are still using the UK date format: **28-11-2019**.

Was this tested and did you confirm that it works?

**Comment by andrewatfornax** [comment]
Hey @420Banners,

Yes, based on the test scenarios that I worked through, the issue was fixed by these changes.

Do you have any users in the system other than the administrator and default manager?

**Comment by 420Banners** [comment]
Yes! 

admin-access.php shows 2 accounts. 
agency-index.php shows a default account. 

We also have some accounts for advertisers but the latest report from an advertiser does not have a user account we created through advertiser-access.php?clientid=???

**Comment by andrewatfornax** [comment]
Okay - please check the language pack preferences for the advertiser user(s) (if they exist) or the relevant manager account in which the advertiser is set up.

The preferences of the "owning" advertiser user (or, if they don't exist, the manager user) are what is used when formatting the email dates, because it is their preference for their emails!

**Comment by 420Banners** [comment]
I log in to the revive adserver control panel, switch from admin to the default account at the top-right corner. The default account lists all of the advertisers in Inventory. 

In /www/admin/account-preferences-timezone.php 
It's set to America/Los_Angeles. 

/www/admin/account-user-name-language.php
Is set to English. 

Am I looking in the correct place? Because I assume those settings would be right to make it display US dates for these types of emails.

**Comment by 420Banners** [comment]
I see a new version has been released but without notes for this issue, so I'm bumping it up.

**Comment by erikgeurts** [comment]
Bumping is not necessary, and it only creates noise.

**Comment by 420Banners** [comment]
With all due respect, we've been trying to get you guys to do this for two years now after a hundred attempts.

 

Maybe a little noise is needed to get this done :)

 

Greenest Regards,

Webmaster

 

420 Magazine

PO Box 3420

Hollywood, CA 90078

http://www.420magazine.com <http://www.420magazine.com/> 

http://www.twitter.com/420magazine

http://www.youtube.com/420magazine 

http://www.facebook.com/420magazine

http://www.instagram.com/420magazine 

 

 

 

Confidentiality Notice. This e-mail is covered by the Electronic Communications Privacy Act, 18 U.S.C. Sections 2510-2521 and is legally privileged. This e-mail is the property of 420 Magazine, Inc. and contains confidential and/or privileged information. This e-mail is intended to be received and reviewed only by the individual named above. Any review, use, dissemination, disclosure or copying of this e-mail or any attachments is strictly prohibited. If you have received this e-mail in error, please immediately notify the sender at support@420magazine.com and delete all copies of the e-mail and any attachments. Thank you for your cooperation and assistance.

**Comment by 420Banners** [comment]
Bump

**Comment by erikgeurts** [comment]
Bumping is not necessary, and it only creates noise.

**Comment by 420Banners** [comment]
We've been bumping this for over a year, how much noise can we make to get this done?

You keep asking for donations, but keep blowing us off.

I'll gladly pay to get this fixed, how much?

**Comment by erikgeurts** [comment]
Extortion sure does not work.

**Comment by 420Banners** [comment]
Well then Mr. Geurts, can you please be a good sport and solve the grand mystery and save us all the trouble of wasting valuable time and energy trying to figure it out by ourselves for a year and a half, by telling us what does? :)

**Comment by 420Banners** [comment]
Yet another 22 days has passed with no reply, what are your thoughts?

**Comment by andrewatfornax** [comment]
That I don't necessarily have time to look at an issue within a 22 day timeframe :-)
Obviously, I want this issue to be resolved - so I will look into it as soon as I possibly can. I am targeting a hopeful return to more active involvement in the project again in Feb.
